templates/java/string.go

```diff
@@ -0,0 +1,78 @@
+package java
+
+const stringTpl = `{{ $f := .Field }}{{ $r := .Rules -}}
+{{- if $r.Const }}
+			com.lyft.pgv.StringValidation.constant("{{ $f.FullyQualifiedName }}", proto.{{ accessor . }}, "{{ $r.GetConst }}");
```

Comment:
I see this code duplicated a lot for different types of fields. The C++ template has a common `in.go` file that is abstract enough to work with all the target types. I don't know if you want to do that here, but it might make sense to combine these into a templated `ValidationHelper.contains<T>` function instead. Same goes for `.in` and .notIn`.

Reply:
I refactored the java stub side to use shared `CollectiveValidation`, `ComparativeValidation`, and `ConstantValidation` classes. 

I haven't yet refactored the generator templates because the java literal syntax is so different for numbers, strings, enums, and WKTs.